content/en/blog/_posts/2023-01-17-confidential-kubernetes.md

```diff
@@ -0,0 +1,152 @@
+
+TPMs are optimized for low cost, allowing them to be integrated into mainboards and act as a system's physical root of trust. To keep the cost low, TPMs are limited in scope, i.e., they provide storage for only a few keys and are capable of just a small subset of cryptographic operations.
+
+In contrast, HSMs are optimized for high performance, so they provide secure storage for far more keys and offer advanced physical attack detection mechanisms. Additionally, high-end HSMs can be programmed such that arbitrary code can be compiled and executed on them. The downside is that they are very costly. A managed CloudHSM from AWS costs [around $1.50 / hour](https://aws.amazon.com/cloudhsm/pricing/) or ~$13,500 / year.
+
+In recent years a new kind of TEE has gained popularity. Technologies like [AMD SEV](https://developer.amd.com/sev/), [Intel SGX](https://www.intel.com/content/www/us/en/developer/tools/software-guard-extensions/overview.html), and [Intel TDX](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-trust-domain-extensions.html) provide TEEs that are closely integrated with userspace. Rather than low-power or high-performance devices that support specific use cases, these TEEs shield normal processes or virtual machines and can do so with relatively low overhead. These technologies each have different design goals, advantages, and limitations, and they are available in different environments including consumer laptops, servers, and mobile devices.
+
+Additionally, we should mention [ARM TrustZone](https://www.arm.com/technologies/trustzone-for-cortex-a) which is optimized for embedded devices such as smartphones, tablets, and smart TVs, as well as [AWS Nitro Enclaves](https://aws.amazon.com/ec2/nitro/nitro-enclaves/) which are only available on [Amazon Web Services](https://aws.amazon.com/) and have a different attacker model compared to the CPU-based solutions by Intel and AMD.
+
+### Security Properties and Feature Set
+
+In the following, we will review the security properties and additional features these new technologies bring to the table. Not all solutions will provide all properties, and we will discuss each technology in further detail in their respective chapter.
+
+The **Confidentiality** property ensures that information cannot be viewed while it is in use in the TEE. This provides us with the highly desired feature to secure **data in use**. Depending on the specific TEE used, both code and data may be protected from outside viewers.
+
+Confidentiality is a great feature, but an attacker would still be able to manipulate or inject arbitrary code and data for the TEE to execute and therefore easily leak critical information. **Integrity** allows a TEE owner to check that neither code nor data has been tempered before running critical computations.
```

Comment:
`Integrity allows a TEE owner to check that neither code nor data has been tempered before running critical computations.` --> `Integrity guarantees a TEE owner that neither code nor data can be tampered with while running critical computations.`

The current wording, particularly where we say that it allows an owner to check the code before a computation, sounds a bit more like attestation to me.